src/screens/transaction-request/qr-code-tx-request/qr-code-tx-request.tsx

```diff
@@ -0,0 +1,627 @@
+    setNetworkTestNetChainId: typeof setNetworkTestNetChainId;
+    toggleTestNet: typeof toggleTestNet;
+    isTestNet: boolean;
+    tokens: ITokensConfigState;
+    setSelectedWallet: typeof setSelectedWallet;
+}
+
+const mapStateToProps = (state: IReduxState) => {
+    const selectedAccount = getSelectedAccount(state);
+
+    return {
+        wallets: state.wallets,
+        selectedWallet: getSelectedWallet(state),
+        selectedAccount,
+        currentChainId: getChainId(state, selectedAccount.blockchain),
+        exchangeRates: state.market.exchangeRates,
+        isTestNet: state.preferences.testNet,
+        tokens: state.tokens
+    };
+};
+
+const mapDispatchToProps = {
+    openBottomSheet,
+    setNetworkTestNetChainId,
+    toggleTestNet,
+    setSelectedWallet
+};
+
+interface IState {
+    toAddress: string;
+    amount: string;
+    chainId: ChainIdType;
+    tokenSymbol: string;
+    insufficientFunds: boolean;
+    insufficientFundsFees: boolean;
+    feeOptions: IFeeOptions;
+    token: ITokenState;
+    isLoading: boolean;
+}
+
+export class QRCodeTransferRequestComponent extends React.Component<
+    IExternalProps & IReduxProps & IThemeProps<ReturnType<typeof stylesProvider>>,
+    IState
+> {
+    constructor(
+        props: IExternalProps & IReduxProps & IThemeProps<ReturnType<typeof stylesProvider>>
+    ) {
+        super(props);
+
+        this.state = {
+            toAddress: '',
+            amount: '',
+            chainId: props.currentChainId,
+            tokenSymbol: getBlockchain(props.selectedAccount.blockchain).config.coin, // Default Native Coin
+            insufficientFunds: false,
+            insufficientFundsFees: false,
+            feeOptions: undefined,
+            token: undefined,
+            isLoading: false
+        };
+    }
+
+    public componentDidMount() {
+        this.parseQrCodeTxPayload();
+    }
+
+    public componentDidUpdate(prevProps: IReduxProps & IExternalProps) {
+        if (this.props.qrCodeTxPayload !== prevProps.qrCodeTxPayload) {
+            this.reinitState();
+        }
+    }
+
+    private reinitState() {
+        this.setState(
+            {
+                amount: '0',
+                chainId: this.props.currentChainId,
+                tokenSymbol: getBlockchain(this.props.selectedAccount.blockchain).config.coin, // Default Native Coin
+                insufficientFunds: false,
+                insufficientFundsFees: false,
+                feeOptions: undefined,
+                token: undefined,
+                isLoading: false
+            },
+            () => this.parseQrCodeTxPayload()
+        );
+    }
+
+    public async parseQrCodeTxPayload() {
+        const { blockchain } = this.props.selectedAccount;
+        const { qrCodeTxPayload } = this.props;
+        const blockchainInstance = getBlockchain(blockchain);
+
+        this.setState({ isLoading: true });
+
+        if (qrCodeTxPayload?.chainId) {
```

Comment:
this is not quite ok...
chainId could be mainnet chainID, and here you are assuming that chainID could be only testnet